File: 프로그래머스/1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.py
def solution(board, h, w):
    answer = 0
    W = len(board[0])
    H = len(board)
    dx = [1,-1,0,0]
    dy = [0,0,1,-1]
    mainColor = board[h][w]
    for i in range(4):
        nx = dx[i] + w
        ny = dy[i] + h
        if 0<=nx<W and 0<=ny<H and board[ny][nx] == mainColor:
            answer +=1
    return answer

File: 프로그래머스/1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.java
class Solution {
    public int solution(String[][] board, int h, int w) {
        int n = board[0].length;
        int count = 0;
        int h_check = 0;
        int w_check = 0;
        int[] dh = {0, 1, -1, 0};
        int[] dw = {1, 0, 0, -1};
        for (int i =0; i<4; i++) {
            h_check = dh[i] + h;
            w_check = dw[i] + w;
            if ((h_check>=0 && h_check<n) && (w_check>=0 && w_check<n)) {
                if (board[h][w].equals(board[h_check][w_check])) {
                    count +=1;
                }
            }
        }
        return count;
    }
}